crates/wit-component/src/printing.rs

```diff
@@ -893,6 +897,54 @@ impl WitPrinter {
         Ok(())
     }
 
+    fn declare_stream(&mut self, resolve: &Resolve, name: Option<&str>, ty: &Type) -> Result<()> {
```

Comment:
Could this take a non-optional name and an error be generated above if `ty.name` is none?

Reply:
I was following the precedent set by `declare_list`, `declare_option`, etc. above, which all take a `Option<&str>`.  Should all of them be changed to non-optional?  Or if not, what's different about `stream`?

Reply:
Ah no I didn't realize that's how everything worked. It feels weird to me to have a method that when it doesn't do anything it's silently a bug, but if it's how everything else is there's no need to change just this method.